File: 0443.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    int compress(vector<char>& chars) {
        char last = ' ';
        int head = 0;
        int count = 1;
        for(int i = 0; i < chars.size(); ++i) {
            if(chars[i] == last) {
                count++;
                continue;
            }
            if(count > 1) {
                int start = head;
                while(count) {
                    chars[head] = count % 10 + '0';
                    head++;
                    count /= 10;
                }
                reverse(chars.begin()+start, chars.begin()+head);
                count = 1;
            }
            last = chars[i];
            chars[head] = chars[i];
            head++;
        }
        if(count > 1) {
            int start = head;
            while(count) {
                chars[head] = count % 10 + '0';
                head++;
                count /= 10;
            }
            reverse(chars.begin()+start, chars.begin()+head);
        }
        chars.resize(head);
        return head;
    }
};

File: 0443.py
class Solution:
    def compress(self, chars: list[str]) -> int:
        last = ""
        head = 0
        count = 1
        for i in range(len(chars)):
            if chars[i] == last:
                count += 1
                continue
            if count > 1:
                for c in list(str(count)):
                    chars[head] = c
                    head += 1
                count = 1
            last = chars[i]
            chars[head] = chars[i]
            head += 1
        if count > 1:
            for c in list(str(count)):
                chars[head] = c
                head += 1
        chars = chars[0:head]
        return head
